tartufo/scanner.py

```
@@ -348,25 +349,36 @@ def config_data(self, data: MutableMapping[str, Any]) -> None:
-        findings = self.config_data.get("exclude_findings", None)
-        if findings:
-            configured_signatures.extend([finding["signature"] for finding in findings])
-
-        self._excluded_findings = tuple(
-            set(self.global_options.exclude_signatures + tuple(configured_signatures))
-        )
-        return self._excluded_findings
+    def excluded_signatures(self) -> tuple:
+        if len(self._excluded_signatures) == 0:
+            signatures = list(self.global_options.exclude_signatures or ())
+            exclude_signatures = list(self.config_data.get("exclude_signatures", ()))
+            try:
+                signatures = signatures + [
+                    x for x in exclude_signatures if x not in signatures
+                ]
+                signatures = [
+                    signature["signature"]
+                    for signature in cast(List[Dict[str, str]], signatures)
+                ]
+                self._excluded_signatures = tuple(cast(List[str], signatures))
+                return self._excluded_signatures
+            except TypeError:
+                warnings.warn(
+                    "Configuring exclude-signatures as string has been deprecated and support for this format will "
+                    "be removed in the future. Please make sure to update your exclude-signatures configuration to "
+                    "an array of tables. For example: exclude-signatures = [{signature='signature', reason='The "
+                    "reason of excluding the signature'}]",
+                    DeprecationWarning,
+                )
+            try:
+                signatures = list(set(signatures + exclude_signatures))
+                self._excluded_signatures = tuple(cast(List[str], signatures))
+            except TypeError as exc:
+                raise types.ConfigException(
+                    f"Combination of old and new format of exclude-signatures is not supported.\n{exc}"
+                )
```

As it stands now, this is never going to be triggered because the only protected code is a simple conversion from a set to a tuple (which is insensitive to the type(s) of the members). I believe it would be better to raise this error here than weird errors later when the members turn out to be weird (due to mixing and matching) -- but that probably means you need to explicitly look at each member and raise this exception `if not isinstance(signature, str)`.